modules/pipeline/services/pipelinesvc/detail.go

```diff
@@ -121,6 +122,10 @@ func (s *PipelineSvc) Detail(pipelineID uint64) (*apistructs.PipelineDetailDTO,
 					Value: task.Inspect.Events,
 				})
 			}
+			// set analyzed yaml task to NoNeedBySystem to simulate db task's behaviour
+			if p.Status.IsStopByUser() && task.Status == apistructs.PipelineStatusAnalyzed {
+				task.Status = apistructs.PipelineStatusNoNeedBySystem
+			}
```

Comment:
is `p.Status.IsStopByUser() ` should change to `p.Status.IsEndStatus()`

Reply:
use `IsStopByUser` is ok.